### PR TITLE
Automatically Run CI and Upload lib to Charmhub when merged to main

### PR DESCRIPTION
**Pull request by kayra1**
# Description

This adds a workflow file to automatically run CI and upload lib to Charmhub when merged to main.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have bumped the version of any required library.

### CODE REVIEW

**Comment by kayra1** [comment]
This will need a CHARMCRAFT_AUTH secret

**Review by gruyaume — APPROVED** [review]
Looks good and I just added the CHARMCRAFT_AUTH secret

File: .github/workflows/update-lib.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: Update ip-router-interface
```
Comment (gruyaume) [inline .github/workflows/update-lib.yaml]: Actually I just realised, why not leverage the common workflow like in [Kubernetes charm libraries](https://github.com/canonical/kubernetes-charm-libraries/blob/main/.github/workflows/publish-libs.yaml):
- canonical/sdcore-github-workflows/.github/workflows/publish-lib.yaml@main

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: .github/workflows/update-lib.yaml (hunk 1 above)
Comment (kayra1) [reply .github/workflows/update-lib.yaml]: Didn't see this example in the repo i looked at. Updated.

File: .github/workflows/update-lib.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            iprouter:
+              - 'lib/charms/ip_router_interface/v0/ip_router_interface.py'
+
+  publish-multus-lib:
```
Comment (gruyaume) [inline .github/workflows/update-lib.yaml]: multus?

File: metadata.yaml [hunk 3]
```diff
@@ -3,13 +3,4 @@ description: |
   Placeholder charm
 summary: |
   Placeholder charm
-website: https://github.com/canonical/ip-router-interface
```
Comment (gruyaume) [inline metadata.yaml]: This change is not related to the PR title or description, please create a separate PR for it.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: .github/workflows/update-lib.yaml (hunk 2 above)
Comment (kayra1) [reply .github/workflows/update-lib.yaml]: oops

File: metadata.yaml (hunk 3 above)
Comment (kayra1) [reply metadata.yaml]: removed

**Review by gruyaume — APPROVED** [review]
